opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/UserManager.java

```diff
@@ -737,12 +745,37 @@ public AuthenticationResponse login(String organizationId, String username, Stri
             }
         }
 
-        OpenCGAResult<User> userOpenCGAResult = getUserDBAdaptor(organizationId).get(username, INCLUDE_ACCOUNT);
+        OpenCGAResult<User> userOpenCGAResult = getUserDBAdaptor(organizationId).get(username, INCLUDE_ACCOUNT_AND_INTERNAL);
         if (userOpenCGAResult.getNumResults() == 1) {
+            User user = userOpenCGAResult.first();
+            if (!ParamConstants.ADMIN_ORGANIZATION.equals(organizationId)) {
+                // Check user is not banned or account is expired
+                if (UserStatus.BANNED.equals(user.getInternal().getStatus().getId())) {
+                    throw CatalogAuthenticationException.userIsBanned(username);
+                }
+                Date date = TimeUtils.toDate(user.getAccount().getExpirationDate());
+                if (date == null) {
+                    throw new CatalogException("Unexpected null expiration date for user '" + username + "'.");
+                }
+                if (date.before(new Date())) {
+                    throw CatalogAuthenticationException.accountIsExpired(username, user.getAccount().getExpirationDate());
+                }
+            }
             authId = userOpenCGAResult.first().getAccount().getAuthentication().getId();
             try {
                 response = authenticationFactory.authenticate(organizationId, authId, username, password);
             } catch (CatalogAuthenticationException e) {
+                if (!ParamConstants.ADMIN_ORGANIZATION.equals(organizationId)) {
+                    // We can only lock the account if it is not the root user
+                    int failedAttempts = userOpenCGAResult.first().getInternal().getFailedAttempts();
+                    ObjectMap updateParams = new ObjectMap(UserDBAdaptor.QueryParams.INTERNAL_FAILED_ATTEMPTS.key(), failedAttempts + 1);
+                    if (failedAttempts >= 4) {
```

Comment:
We are not aware of any software that let's you decide the maximum number of login attempts before blocking an account, so that is going to be the number of times a user is allowed to make a login attempt before blocking the account. Obviously, we will review this decision depending on how users react.

Reply:
Also, check:
`pwdMaxFailure` 
https://www.openldap.org/doc/admin24/guide.html

Reply:
Or...
> `deny` – used to define the number of attempts (3 in this case), after which the user account should be locked.

https://www.tecmint.com/lock-failed-ssh-login-attempts-linux/#How_to_Lock_User_After_Failed_SSH_Logins